subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/metrics/model/DeviceTimestamps.kt

```diff
@@ -0,0 +1,25 @@
+package com.avito.runner.scheduler.metrics.model
+
+import com.avito.math.percentOf
+
+internal data class DeviceTimestamps(
+    val created: Long,
+    val testTimestamps: MutableMap<TestKey, TestTimestamps>,
+    val finished: Long
+) {
+
+    private val totalTime
+        get() = finished - created
+
+    private val effectiveWorkTime
+        get() = testTimestamps.values.mapNotNull { it.effectiveWorkTime }.sum()
+
+    val utilizationPercent: Int?
+        get() = if (totalTime > effectiveWorkTime) {
+            effectiveWorkTime.percentOf(totalTime).toInt()
+        } else {
+            null // in case if device died and finish time not logged
```

Comment:
> finish time not logged 

How is that possible when finished is a non-null type `val finished: Long` ?

Reply:
total time will be 0 in this case

Reply:
i will make it nullable for consistency